src/pages/dashboard/DashboardPage.tsx

```diff
@@ -25,50 +25,46 @@ const DashboardPage = () => {
   const [ecNodeData, setEcNodeData] = useState<CreditorInstitutionDetailsResource>();
 
   useEffect(() => {
-    if (
-      selectedParty !== undefined &&
-      selectedParty.pspData !== undefined &&
-      selectedParty.institutionType === 'PSP'
-    ) {
-      getPSPDetails(selectedParty.fiscalCode)
-        .then((response) => {
-          setPspNodeData(response);
-        })
-        .catch((reason) =>
-          addError({
-            id: 'DASHBOARD_PAGE_PSP_DETAILS',
-            blocking: false,
-            error: reason as Error,
-            techDescription: `An error occurred while getting psp details`,
-            toNotify: true,
-            displayableTitle: t('dashboardPage.registrationData.pspDetailsErrorMessageTitle'),
-            displayableDescription: t('dashboardPage.registrationData.pspDetailsErrorMessageDesc'),
-            component: 'Toast',
+    if (selectedParty !== undefined) {
```

Comment:
you can simply write selectedParty inside of the condition, in this way it is cleaner and, in addition to the diversity from undefined, also that from null is checked